src/new_index/fetch.rs

```diff
@@ -165,7 +186,16 @@ fn blkfiles_reader(blk_files: Vec<PathBuf>, xor_key: Option<[u8; 8]>) -> Fetcher
     Fetcher::from(
         chan.into_receiver(),
         spawn_thread("blkfiles_reader", move || {
+            let mut count = 0;
+            let blk_files_len = blk_files.len();
             for path in blk_files {
```

Comment:
nit: usually `enumerate()` is a more idiomatic way to do it

Reply:
added addition commit with this suggestion, thank you, please review

Reply:
squashed